core/src/main/java/org/jdbi/v3/core/StatementContext.java

```diff
@@ -155,6 +157,13 @@ public Object getAttribute(String key)
         return config.collectorRegistry.findCollectorFor(type);
     }
 
+    public <C extends ExtensionConfig<C>> C findConfiguration(Class<C> configType)
```

Comment:
This is kinda weird too.  Do we actually expect to have multiple configurations loaded ever?  Shouldn't just the most specific one win?  I'm not sure the `configure(Class, Consumer)` API makes sense, why not just `getConfig(Class)` that returns the most specific one.